pkg/server/message.go

```diff
@@ -254,69 +271,145 @@ func (t *TranslateServiceServer) UpdateMessages(
 
 // helpers
 
-// populateTranslatedMessages adds any missing messages from the original messages to all translated messages.
-func (t *TranslateServiceServer) populateTranslatedMessages(
-	ctx context.Context,
-	serviceID uuid.UUID,
-	originalMessages *model.Messages,
-	allMessages []model.Messages,
-) error {
+// alterTranslations alter the status of specific messages in multiple languages to "untranslated"
+// based on a list of message IDs, while keeping the "original" language messages unchanged.
+func (t *TranslateServiceServer) alterTranslations(
+	all model.MessagesSlice,
+	untranslatedIDs []string,
+) model.MessagesSlice {
+	if len(untranslatedIDs) == 0 || len(all) == 1 {
+		return all
+	}
+
+	original, others := all.SplitOriginal()
+
+	// Return if messages don't contain original language
+	if original == nil {
+		return all
+	}
+
+	slices.Sort(untranslatedIDs)
+
+	// Update altered messages for all translations
+	for _, msg := range others {
+		for i := range msg.Messages {
+			if _, found := slices.BinarySearch(untranslatedIDs, msg.Messages[i].ID); found {
+				msg.Messages[i].Status = model.MessageStatusUntranslated
+			}
+		}
+	}
+
+	return all
+}
+
+// populateTranslations adds any missing messages from the original messages to all translated messages,
+// returns messagesSlice containing populated translations.
+func (t *TranslateServiceServer) populateTranslations(all model.MessagesSlice) []model.Messages {
```

Comment:
Does this function work? The logic in this function is not correct, the lookup contains the same messages as in the `for` loop. It compares to the same `Messages` structure.

Reply:
Tests were way off :)
Fixed the tests, and now I am sure that everything is working